utils/common/collation.h

```diff
@@ -186,6 +186,30 @@ class Charset
                                    '\\','_','%');
       return neg ? !res : res;
     }
+    size_t strnxfrm(uchar *dst, size_t dstlen, uint nweights,
+                    const uchar *src, size_t srclen, uint flags)
+    {
+        if (mCharset->coll)
+        {
+            return mCharset->coll->strnxfrm(mCharset, dst, dstlen, nweights, src, srclen, flags);
+        }
+        else
+        {
+            // when collation is not specified, we use string as it is.
+            // this is more or less "safe" fallback, put here "just in case"
+            size_t i;
+            if (dstlen > nweights)
+            {
+                dstlen = nweights;
+            }
+            for (i = 0; i < dstlen; i ++)
+            {
+                uint8_t ch = i < srclen ? src[i] : 0;
```

Comment:
Since MCS 6.2.1 there is an explicit requirement for SSE4.2 in a runtime so could you plz:
- divide the loop in two where first processes from 0 to min(dstLen, srcLen) and second fills up a slice dst[dstLen - srcLen:-1] with 0
- vectorize the first loop if the size of processable common prefix >= 16 using the vectorization [patch](https://github.com/mariadb-corporation/mariadb-columnstore-engine/commit/dafb9c5daece876dd991b4eb63170f0a1de9ecf5) as the example (feel free to ask any questions)?